app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  def index
+    users = User.all
+
+    render json: users, status: 200
+  end
+
+  def show
+    user = User.find(params[:id])
+
+    render json: user, status: 200
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if user.save!
+      render json: user, status: 201
+    else
+      render json: {error: "User is invalid"}, status: 400
```

Comment:
the statement under ```else``` will be ran if the ```POST``` request doesn't have valid fields for ```user``` model. The 400 error is a bad request

 ```The server cannot or will not process the request due to an apparent client error (e.g., malformed request syntax, too large size, invalid request message framing, or deceptive request routing)```

I thought this was the best fitting. No?